eclipse-theia/theia
# notebook kernel messaging and renderer messaging fixes

#### What it does

This implements the proposed Kernel messaging API (`vscode.proposed.notebookMessaging.d.ts`) and the `notebookPreload` contribution point. Also fixes notebook renderer messaging. This is to support interactive output widgets like for the `ipywidgets` library.

Tested with ipywidgets 7.8.1 and 8.1.2

#### How to test

1. `pip install ipywidgets` in your used kernel
2. Open a notebook
3. add following cell
```
import ipywidgets as widgets
widgets.IntSlider(min=1, max=12)
```
4. Execute the cell and see the interactive slider is rendered

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
